# Read dataset into a dataframe

In [10]:
import pandas as pd
df = pd.read_csv("/content/household_power_clean.csv")

In [11]:
df.head()

Unnamed: 0        Date      Time  Global_active_power  \
0           0  2006-12-16  17:24:00                4.216   
1           1  2006-12-16  17:25:00                5.360   
2           2  2006-12-16  17:26:00                5.374   
3           3  2006-12-16  17:27:00                5.388   
4           4  2006-12-16  17:28:00                3.666   

   Global_reactive_power  Voltage  Global_intensity  Sub_metering_1  \
0                  0.418   234.84              18.4             0.0   
1                  0.436   233.63              23.0             0.0   
2                  0.498   233.29              23.0             0.0   
3                  0.502   233.74              23.0             0.0   
4                  0.528   235.68              15.8             0.0   

   Sub_metering_2  Sub_metering_3             Datetime  gap_monthly  \
0             1.0            17.0  2006-12-16 17:24:00          NaN   
1             1.0            16.0  2006-12-16 17:25:00          NaN   
2             2.0            17.0  2006-12-16 17:26:00          NaN   
3             1.0            17.0  2006-12-16 17:27:00          NaN   
4             1.0            17.0  2006-12-16 17:28:00          NaN   

   grp_monthly  v_monthly  gi_monthly  
0          NaN        NaN         NaN  
1          NaN        NaN         NaN  
2          NaN        NaN         NaN  
3          NaN        NaN         NaN  
4          NaN        NaN         NaN

In [12]:
df.tail()

Unnamed: 0        Date      Time  Global_active_power  \
2049275     2075254  2010-11-26  20:58:00                0.946   
2049276     2075255  2010-11-26  20:59:00                0.944   
2049277     2075256  2010-11-26  21:00:00                0.938   
2049278     2075257  2010-11-26  21:01:00                0.934   
2049279     2075258  2010-11-26  21:02:00                0.932   

         Global_reactive_power  Voltage  Global_intensity  Sub_metering_1  \
2049275                    0.0   240.43               4.0             0.0   
2049276                    0.0   240.00               4.0             0.0   
2049277                    0.0   239.82               3.8             0.0   
2049278                    0.0   239.70               3.8             0.0   
2049279                    0.0   239.55               3.8             0.0   

         Sub_metering_2  Sub_metering_3             Datetime  gap_monthly  \
2049275             0.0             0.0  2010-11-26 20:58:00     0.971133   
2049276             0.0             0.0  2010-11-26 20:59:00     0.970667   
2049277             0.0             0.0  2010-11-26 21:00:00     0.969267   
2049278             0.0             0.0  2010-11-26 21:01:00     0.967667   
2049279             0.0             0.0  2010-11-26 21:02:00     0.966000   

         grp_monthly   v_monthly  gi_monthly  
2049275       0.0348  238.936667    4.080000  
2049276       0.0348  238.975000    4.080000  
2049277       0.0348  238.988000    4.066667  
2049278       0.0348  239.000333    4.053333  
2049279       0.0348  238.978667    4.040000

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2049280 entries, 0 to 2049279
Data columns (total 15 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   Unnamed: 0             int64  
 1   Date                   object 
 2   Time                   object 
 3   Global_active_power    float64
 4   Global_reactive_power  float64
 5   Voltage                float64
 6   Global_intensity       float64
 7   Sub_metering_1         float64
 8   Sub_metering_2         float64
 9   Sub_metering_3         float64
 10  Datetime               object 
 11  gap_monthly            float64
 12  grp_monthly            float64
 13  v_monthly              float64
 14  gi_monthly             float64
dtypes: float64(11), int64(1), object(3)
memory usage: 234.5+ MB


In [14]:
df.columns

Index(['Unnamed: 0', 'Date', 'Time', 'Global_active_power',
       'Global_reactive_power', 'Voltage', 'Global_intensity',
       'Sub_metering_1', 'Sub_metering_2', 'Sub_metering_3', 'Datetime',
       'gap_monthly', 'grp_monthly', 'v_monthly', 'gi_monthly'],
      dtype='object')

In [15]:
df.describe()

Unnamed: 0  Global_active_power  Global_reactive_power       Voltage  \
count  2.049280e+06         2.049280e+06           2.049280e+06  2.049280e+06   
mean   1.031496e+06         1.091615e+00           1.237145e-01  2.408399e+02   
std    5.964586e+05         1.057294e+00           1.127220e-01  3.239987e+00   
min    0.000000e+00         7.600000e-02           0.000000e+00  2.232000e+02   
25%    5.162538e+05         3.080000e-01           4.800000e-02  2.389900e+02   
50%    1.028638e+06         6.020000e-01           1.000000e-01  2.410100e+02   
75%    1.545307e+06         1.528000e+00           1.940000e-01  2.428900e+02   
max    2.075258e+06         1.112200e+01           1.390000e+00  2.541500e+02   

       Global_intensity  Sub_metering_1  Sub_metering_2  Sub_metering_3  \
count      2.049280e+06    2.049280e+06    2.049280e+06    2.049280e+06   
mean       4.627759e+00    1.121923e+00    1.298520e+00    6.458447e+00   
std        4.444396e+00    6.153031e+00    5.822026e+00    8.437154e+00   
min        2.000000e-01    0.000000e+00    0.000000e+00    0.000000e+00   
25%        1.400000e+00    0.000000e+00    0.000000e+00    0.000000e+00   
50%        2.600000e+00    0.000000e+00    0.000000e+00    1.000000e+00   
75%        6.400000e+00    0.000000e+00    1.000000e+00    1.700000e+01   
max        4.840000e+01    8.800000e+01    8.000000e+01    3.100000e+01   

        gap_monthly   grp_monthly     v_monthly    gi_monthly  
count  2.049251e+06  2.049251e+06  2.049251e+06  2.049251e+06  
mean   1.091592e+00  1.237133e-01  2.408399e+02  4.627662e+00  
std    9.516172e-01  7.736539e-02  3.080626e+00  3.987975e+00  
min    8.166667e-02  0.000000e+00  2.255873e+02  2.200000e-01  
25%    3.301333e-01  6.960000e-02  2.391050e+02  1.460000e+00  
50%    7.143333e-01  1.030000e-01  2.409840e+02  3.073333e+00  
75%    1.569933e+00  1.593333e-01  2.427670e+02  6.560000e+00  
max    7.966200e+00  7.897333e-01  2.524423e+02  3.423333e+01

# Check for null values and drop them

In [16]:
df.isnull().sum()

Unnamed: 0                0
Date                      0
Time                      0
Global_active_power       0
Global_reactive_power     0
Voltage                   0
Global_intensity          0
Sub_metering_1            0
Sub_metering_2            0
Sub_metering_3            0
Datetime                  0
gap_monthly              29
grp_monthly              29
v_monthly                29
gi_monthly               29
dtype: int64

In [17]:
df = df.dropna()

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2049251 entries, 29 to 2049279
Data columns (total 15 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   Unnamed: 0             int64  
 1   Date                   object 
 2   Time                   object 
 3   Global_active_power    float64
 4   Global_reactive_power  float64
 5   Voltage                float64
 6   Global_intensity       float64
 7   Sub_metering_1         float64
 8   Sub_metering_2         float64
 9   Sub_metering_3         float64
 10  Datetime               object 
 11  gap_monthly            float64
 12  grp_monthly            float64
 13  v_monthly              float64
 14  gi_monthly             float64
dtypes: float64(11), int64(1), object(3)
memory usage: 250.2+ MB


In [19]:
df.isnull().sum()

Unnamed: 0               0
Date                     0
Time                     0
Global_active_power      0
Global_reactive_power    0
Voltage                  0
Global_intensity         0
Sub_metering_1           0
Sub_metering_2           0
Sub_metering_3           0
Datetime                 0
gap_monthly              0
grp_monthly              0
v_monthly                0
gi_monthly               0
dtype: int64

# Take last 10000 rows of the dataset as Tableau dashboard cannot take the entire dataset. Note that the original dataset (after removing null values) has 2049251 rows.

In [23]:
subset_df = df.tail(10000)

In [24]:
subset_df.shape

(10000, 15)

# Save the subset to a CSV file

In [25]:
# Save the subset
subset_file_path = "/content/household_power_clean_subset.csv"
subset_df.to_csv(subset_file_path, index=False)

print(f"Subset saved as {subset_file_path}")

Subset saved as /content/household_power_clean_subset.csv


In [26]:
subset_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10000 entries, 2039280 to 2049279
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0             10000 non-null  int64  
 1   Date                   10000 non-null  object 
 2   Time                   10000 non-null  object 
 3   Global_active_power    10000 non-null  float64
 4   Global_reactive_power  10000 non-null  float64
 5   Voltage                10000 non-null  float64
 6   Global_intensity       10000 non-null  float64
 7   Sub_metering_1         10000 non-null  float64
 8   Sub_metering_2         10000 non-null  float64
 9   Sub_metering_3         10000 non-null  float64
 10  Datetime               10000 non-null  object 
 11  gap_monthly            10000 non-null  float64
 12  grp_monthly            10000 non-null  float64
 13  v_monthly              10000 non-null  float64
 14  gi_monthly             10000 non-null  float64
dtyp

In [27]:
subset_df.describe()

Unnamed: 0  Global_active_power  Global_reactive_power       Voltage  \
count  1.000000e+04         10000.000000           10000.000000  10000.000000   
mean   2.070258e+06             1.146602               0.093189    240.664805   
std    2.886896e+03             1.003507               0.087506      3.580226   
min    2.065259e+06             0.210000               0.000000    225.320000   
25%    2.067759e+06             0.326000               0.048000    238.180000   
50%    2.070258e+06             0.875000               0.070000    241.060000   
75%    2.072758e+06             1.586000               0.134000    243.440000   
max    2.075258e+06             9.724000               0.802000    249.100000   

       Global_intensity  Sub_metering_1  Sub_metering_2  Sub_metering_3  \
count      10000.000000    10000.000000    10000.000000    10000.000000   
mean           4.847320        1.434500        0.972400        6.124300   
std            4.283948        6.559528        4.802952        8.340864   
min            1.000000        0.000000        0.000000        0.000000   
25%            1.400000        0.000000        0.000000        0.000000   
50%            3.800000        0.000000        0.000000        0.000000   
75%            6.600000        0.000000        1.000000       17.000000   
max           43.000000       57.000000       69.000000       19.000000   

        gap_monthly   grp_monthly     v_monthly    gi_monthly  
count  10000.000000  10000.000000  10000.000000  10000.000000  
mean       1.146399      0.093239    240.661549      4.846564  
std        0.901647      0.052301      3.385727      3.843827  
min        0.241200      0.019067    228.661000      1.080000  
25%        0.334133      0.058667    238.245833      1.433333  
50%        1.029000      0.075467    241.021000      4.320000  
75%        1.644000      0.115933    243.322250      6.933333  
max        7.216667      0.360333    247.087000     31.820000

# Save the file as HTML and PDF

## Install required libraries to convert to a HTML and PDF

In [28]:
!apt-get update
!apt-get install pandoc
!apt-get install inkscape
!apt-get install texlive-xetex texlive-fonts-recommended texlive-latex-extra

Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]
Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]
Get:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]
Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease
Get:5 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]
Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]
Get:7 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ Packages [66.7 kB]
Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,309 kB]
Get:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]
Hit:10 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease
Get:11 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease [24.3 kB]
Get:12 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,663 kB]
Hit:13 https://ppa.launchpadcont

## Mount Google Drive

In [29]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## Command to convert the file to HTML

In [35]:
%shell jupyter nbconvert --to html '/content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.ipynb'

[NbConvertApp] Converting notebook /content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.ipynb to html
[NbConvertApp] Writing 398526 bytes to /content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.html


## Command to convert the file to PDF

In [36]:
%shell jupyter nbconvert --to pdf '/content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.ipynb'

[NbConvertApp] Converting notebook /content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.ipynb to pdf
[NbConvertApp] Writing 108982 bytes to notebook.tex
[NbConvertApp] Building PDF
[NbConvertApp] Running xelatex 3 times: ['xelatex', 'notebook.tex', '-quiet']
[NbConvertApp] Running bibtex 1 time: ['bibtex', 'notebook']
[NbConvertApp] WARNING | bibtex had problems, most likely because there were no citations
[NbConvertApp] PDF successfully created
[NbConvertApp] Writing 81239 bytes to /content/drive/MyDrive/Colab Notebooks/Ravi_Teja_Kothuru_AAI_530_A1_M5_Assignment_5.1.pdf
